### DISCUSSION TITLE
Formatting

### DISCUSSION THREAD

**Discussion by Decodetalkers** [ebkalderon/tower-lsp#354] [Q&A] [answered]
How to set the format option when using formattiong function?

**Comment by ebkalderon** [comment, marked as answer]
What sort of formatting options do you mean? The `LanguageServer::formatting()` method has a [`DocumentFormattingParams`](https://docs.rs/lsp-types/latest/lsp_types/struct.DocumentFormattingParams.html) argument passed into it, and this struct includes a [`FormattingOptions`](https://docs.rs/lsp-types/latest/lsp_types/struct.FormattingOptions.html) member that contains a bunch of fields describing to the server how the formatting should be applied. Is this what you meant by your question?
